Add CI infrastructure

apmasell opened the pull request:
Copy configuration from numba-scipy

apmasell committed:
Add CI infrastructure

Copy configuration from numba-scipy

apmasell committed:
Fix formatting

Make formatting match flake8

sklam commented on .github/workflows/test_and_deploy.yml:
```diff
@@ -58,47 +57,61 @@ jobs:
-        pyver:
-          - 3.7
-          - 3.8
+        pyver: [3.6, 3.7, 3.8]
```
Latest numba doesn't support 3.6. And let's add new python versions
```suggestion
        pyver: [3.7, 3.8, 3.9, 3.10]
```

sklam commented on .github/workflows/test_and_deploy.yml:
```diff
@@ -58,47 +57,61 @@ jobs:
+          ls -la ./artifact_storage
+          UPLOAD=$(ls ./artifact_storage | head -1)
+          echo "Uploading $UPLOAD with label=${{ env.AC_LABEL }}"
+          $CONDA/bin/anaconda -t ${{ secrets.ANACONDA_ORG_TOKEN }} upload -u numba -l ${{ env.AC_LABEL }} --no-progress --force --no-register ./artifact_storage/$UPLOAD
```
i've added the secret token

apmasell replied:
Thank you.

sklam commented on .github/workflows/deploy_release.yml:
```diff
@@ -0,0 +1,35 @@
+        uses: pypa/gh-action-pypi-publish@v1.4.2
+        with:
+          user: __token__
+          password: ${{ secrets.PYPI_SECRET }}
```
Not sure if we should publish to PYPI just yet. We should probably see how it works on anaconda.org first.

sklam commented on .github/workflows/test_and_deploy.yml:
```diff
@@ -58,47 +57,61 @@ jobs:
       - name: Pytest
+        shell: bash -l {0}
+        run: |
+          pytest --pyargs numba_extras.tests
```
Testing for this repo is a bit different. Each submodule will be self-contained.

```suggestion
          pytest --pyargs numba_extras
```

apmasell committed:
Remove PyPI deployment, fix Python versions, change tests

Remove workflow to submit to PyPI as this isn't mature enough to be useful yet.
Adjust Python versions to test against to match current Numba versions. Change
test scanning package to allow for tests to reside within each feature.

sklam commented:
going to merge it to see if deploy actually works

esc commented:
@sklam how did it turn out?
